docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+  - `num_bins`: int, optional, default=23. Number of bins for intensity.
+  - `sigma_ratio`: float, optional, default=0.5. A hyper parameter for the gaussian
+    function.
+
+#### Label
+
+The label loss calculates dissimilarity between labels.
+
+All pre-implemented losses can be used as multi-scale or single scale losses.
+Multi-scale losses require a kernel Additionally, all losses can be weighted, so the
+following two arguments are global to all provided losses:
+
+- `weight`: float type, weight of individual loss element in total loss function.
+- `scales`: list, or None. Optional argument. If you do not pass this argument (or pass
+  the list [0], the value `null` or an empty value pair), the loss is calculated at a
+  single scale. If you past a list of length > 1, a multi-scale loss will be used.
+  WARNING: an empty list ([]) will raise an error.
+- `kernel`: str, "gaussian" or "cauchy", default "gaussian". Optional argument. Defines
+  the kernel to use for multi-scale losses.
+
+EG.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    label:
+      weight: 1.0
+      name: "dice" # options include "dice", "cross-entropy", "mean-squared", "generalised_dice" and "jaccard"
+      scales: [1, 2]
+```
+
+The default losses require the following arguments. Additional arguments should be added
+at the same indent level:
+
+- `dice`: Calls a Dice loss on the labels, requires the following arguments:
+
+  - `binary`: bool, default is false. If true, the tensors are thresholded at 0.5.
+  - `neg_weight`: float, default=0.0. Defines the weight of the background class (0)
+    versus foreground class, which is weighted as (1-neg_weight).
+
+- `cross-entropy`: Calls a cross-entropy loss between labels, requires the following
+  arguments:
+
+  - `binary`: bool, default is false. If true, the tensors are thresholded at 0.5.
+  - `neg_weight`: float, default=0.0. Defines the weight of the background class (0)
+    versus foreground class, which is weighted as (1-neg_weight).
+
+- `jaccard`: - `binary`: bool, default is false. If true, the tensors are thresholded at
+0.5.
+<!-- - `neg_weight`: float, default=0.0. Defines the weight of the background class (0) versus foreground class, which is weighted as (1-neg_weight). -->
+
+#### Regularization
+
+The regularization section configures the losses for the DDF. To instantiate this part
+of the loss, pass "regularization" into the config file as a field.
+
+- `weight`: float type, weight of the regularization loss.
+- `name`: string type, the type of deformation energy to compute. Options include
+  "bending", "gradient"
+
+If the `gradient` loss is used, another optional argument can be passed at the same
```

Comment:
why "another"?

Reply:
It's not optional, actually. Since get_config will return error as it calls "l1" from the dict. Nice catch.